server/proxy/map_uri.js

```diff
@@ -41,7 +44,19 @@ export default function mapUri(server, mappings) {
 
     const mappedUrl = formatUrl(mappedUrlComponents);
     server.log(['plugin:own-home', 'debug'], 'mappedUrl: ' + mappedUrl);
-    done(null, mappedUrl, request.headers);
+
+    const jwtToken = getJwt(request);
+    if (jwtToken) {
+      verifyJwt(server, jwtToken,
+        () => {
+          done(null, mappedUrl, request.headers);
+        },
+        () => {
+          done(Boom.forbidden('Unauthorized'), null, null);
+        });
+    } else {
+      done(null, mappedUrl, request.headers);
+    }
```

Comment:
Is it possible to change the above lines to
```
    if (config.get('own_home.jwt.enabled')) {
      const jwtToken = getJwt(request);
      verifyJwt(server, jwtToken,
        () => {
          done(null, mappedUrl, request.headers);
        },
        () => {
          done(Boom.forbidden('Unauthorized'), null, null);
        });
    } else {
      done(null, mappedUrl, request.headers);
    }
```
I think this is better in case we add more methods in the future.

Reply:
good catch.